src/web_service.rs

```diff
@@ -331,20 +316,23 @@ pub async fn publish(channel: PublishChannel, value: serde_json::Value) -> Resul
             continue;
         };
 
-        debug!("before");
-
+        /*
+           ToDo: we faced situations when 'request.send()' didn't return after twin::handle_connection_status() was
+           called as a result of 'UnauthenticatedReason::ExpiredSasToken'. As a workaround we use a timeout.
+           The crazy point is that with the timeout the problem disappeared.
+        */
         match tokio::time::timeout_at(
             tokio::time::Instant::now() + tokio::time::Duration::from_secs(15),
             request.send(),
         )
         .await
         {
-            Err(_) => error!("timeout"),
+            Err(_) => bail!("publish_to_endpoints: timeout while sending {msg} to {}.", endpoint.url),
             Ok(Err(e)) => warn!(
-                "publish: sending request {msg} to {} failed with {e}. Endpoint not present?",
+                "publish_to_endpoints: sending {msg} to {} (failed with {e}). Endpoint not present?",
```

Comment:
How is a timeout different from other types of errors that may occure after the connection was established? Why bail in the one case and log an error in the other?